main.go

```diff
@@ -128,10 +341,161 @@ func determineShouldFetch(lastFetchedFilepath string) bool {
 	noFetchNeededBefore := lastFetchedTime.Add(fetchGracePeriod)
 
-	return time.Now().After(noFetchNeededBefore)
+	return time.Now().After(noFetchNeededBefore), nil
 }
+
+func getLatestReleaseVersion(repo *git.Repository) (*semver.Version, error) {
+	tagrefs, err := repo.Tags()
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred while retrieving tags for repository.")
+	}
+
+	// Trim tagrefs and filter for only tags with X.Y.Z version format
+	var allTagSemVers []*semver.Version
+	err = tagrefs.ForEach(func(tagref *plumbing.Reference) error {
+		tagName := tagref.Name().String()
+		tagName = strings.ReplaceAll(tagName, tagsPrefix, "")
+
+		if semverRegex.Match([]byte(tagName)) {
+			tagSemVer, err := semver.StrictNewVersion(tagName)
+			if err != nil {
+				return stacktrace.Propagate(err, "An error occurred while retrieving the following tag: %s.", tagName)
+			}
+			allTagSemVers = append(allTagSemVers, tagSemVer) 
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var latestReleaseTagSemVer *semver.Version
+	if len(allTagSemVers) == 0 {
+		latestReleaseTagSemVer, err = semver.StrictNewVersion(noPreviousVersion)
+		if err != nil {
+			return nil, stacktrace.Propagate(err, "An error occurred creating '%s' semantic version.", noPreviousVersion)
+		}
+	} else {
+		sort.Sort(sort.Reverse(semver.Collection(allTagSemVers)))
+		latestReleaseTagSemVer = allTagSemVers[0]
+	}
+
+	return latestReleaseTagSemVer, nil
+}
+
+func doBreakingChangesExist(changelogFilepath string) (bool, error) {
+	changelogFile, err := os.Open(changelogFilepath)
+	if err != nil {
+		return false, stacktrace.Propagate(err, "An error occurred attempting to open changelog file at provided path. Are you sure '%s' exists?", changelogFilepath)
+	}
+	defer changelogFile.Close()
+
+    scanner := bufio.NewScanner(changelogFile)
+	// Find TBD header
+    for scanner.Scan() {
+		if tbdHeaderRegex.Match(scanner.Bytes()) {
+			break
+		}
+    }
+	if err := scanner.Err(); err != nil {
+        return false, stacktrace.Propagate(err, "An error occurred while scanning for the TBD header in the changelog file at provided path: '%s'.\n", changelogFilepath)
+    }
+	// Scan file until next version header detected, searching for Breaking Changes header along the way
+	foundBreakingChanges := false
+	for scanner.Scan() {
+		if breakingChangesRegex.Match(scanner.Bytes()){
+			foundBreakingChanges = true
+		}
+		if versionHeaderRegex.Match(scanner.Bytes()){
+			break
+		}
+	}
+    if err := scanner.Err(); err != nil {
+		return false, stacktrace.Propagate(err, "An error occurred while scanning for the breaking changes header in the changelog file at provided path: '%s'.\n", changelogFilepath)
+    }
+
+    return foundBreakingChanges, nil
+}
+
+func runPreReleaseScripts(preReleaseScriptsDirpath string, releaseVersion string) error {
+	preReleaseScriptsFilepath := path.Join(preReleaseScriptsDirpath, preReleaseScriptsFilename)
```

Comment:
Ah, the `.pre-release-scripts.txt` should be in the root of the repo (cleaner than expecting `scripts` to exist, which it may or may not)